app/upgrades/v15/upgrades.go

```diff
@@ -66,31 +114,190 @@ func UpgradeSigningInfos(ctx sdk.Context, sk slashingkeeper.Keeper) {
+	dels := sk.GetDelegatorDelegations(ctx, delegator, 100)
+
+	for _, del := range dels {
+		valAddr := del.GetValidatorAddr()
+
+		validator, found := sk.GetValidator(ctx, valAddr)
+		if !found {
+			return stakingtypes.ErrNoValidatorFound
+		}
+
+		returnAmount, err := sk.Unbond(ctx, delegator, valAddr, del.GetShares())
+		if err != nil {
+			return err
+		}
+
+		// transfer the validator tokens to the not bonded pool
+		if validator.IsBonded() {
+			// doing stakingKeeper.bondedTokensToNotBonded
+			coins := sdk.NewCoins(sdk.NewCoin(sk.BondDenom(ctx), returnAmount))
+			err = bk.SendCoinsFromModuleToModule(ctx, stakingtypes.BondedPoolName, stakingtypes.NotBondedPoolName, coins)
+			if err != nil {
+				return err
 			}
-			val.Commission.UpdateTime = ctx.BlockHeader().Time
-			sk.SetValidator(ctx, val)
+		}
+
+		bondDenom := sk.GetParams(ctx).BondDenom
+		amt := sdk.NewCoin(bondDenom, returnAmount)
+
+		err = bk.UndelegateCoinsFromModuleToAccount(ctx, stakingtypes.NotBondedPoolName, delegator, sdk.NewCoins(amt))
+		if err != nil {
+			return err
 		}
 	}
-	ctx.Logger().Info("Finished migrating min commission rate")
+
+	return nil
+}
+
+// forceFundCommunityPool sends the given coin from the sender account to the community pool
+// even if the coin is locked.
+// Note that it partially follows the logic of the FundCommunityPool method in
+// https://github.com/cosmos/cosmos-sdk/blob/release%2Fv0.47.x/x/distribution/keeper/keeper.go#L155
+func forceFundCommunityPool(
+	ak accountkeeper.AccountKeeper,
+	dk distributionkeeper.Keeper,
+	bk bankkeeper.Keeper,
+	ctx sdk.Context,
+	amount sdk.Coin,
+	sender sdk.AccAddress,
+	bs storetypes.StoreKey,
+) error {
+	recipientAcc := ak.GetModuleAccount(ctx, distributiontypes.ModuleName)
+	if recipientAcc == nil {
+		return fmt.Errorf("%s:%s", sdkerrors.ErrUnknownAddress, distributiontypes.ModuleName)
+	}
+
+	senderBal := bk.GetBalance(ctx, sender, amount.Denom)
+	if _, hasNeg := sdk.NewCoins(senderBal).SafeSub(amount); hasNeg {
+		return fmt.Errorf(
+			"%s: spendable balance %s is smaller than %s",
+			sdkerrors.ErrInsufficientFunds,
+			senderBal,
+			amount,
+		)
+	}
+	if err := setBalance(ctx, sender, senderBal.Sub(amount), bs); err != nil {
+		return err
+	}
+	recipientBal := bk.GetBalance(ctx, recipientAcc.GetAddress(), amount.Denom)
+	if err := setBalance(ctx, recipientAcc.GetAddress(), recipientBal.Add(amount), bs); err != nil {
+		return err
+	}
+
+	accExists := ak.HasAccount(ctx, recipientAcc.GetAddress())
+	if !accExists {
+		ak.SetAccount(ctx, ak.NewAccountWithAddress(ctx, recipientAcc.GetAddress()))
+	}
+
+	feePool := dk.GetFeePool(ctx)
+	feePool.CommunityPool = feePool.CommunityPool.Add(sdk.NewDecCoinsFromCoins(amount)...)
+	dk.SetFeePool(ctx, feePool)
+
+	return nil
+}
+
+// setBalance sets the coin balance for an account by address.
+// Note that it follows the same logic of the sendBalance method in
+// https://github.com/cosmos/cosmos-sdk/blob/release%2Fv0.47.x/x/bank/keeper/send.go#L337
```

Comment:
```suggestion
// https://github.com/cosmos/cosmos-sdk/blob/v0.47.7/x/bank/keeper/send.go#L337
```
Otherwise the line may change once release/0.47.x is modified.